lib/client.js

```diff
@@ -932,6 +932,13 @@ function onSocketError (err) {
 }
 
 function onSocketEnd () {
+  const { [kParser]: parser } = this
+
+  if (this.statusCode && !this.shouldKeepAlive) {
+    // Response does not contain content-length nor chunked encoding.
+    parser.onMessageComplete()
+  }
```

Comment:
This might or might not be an expected end of message. We have no way to tell at the moment?